### PR TITLE
EMailTemplate: Fix button background in Lotus Notes

### PR DESCRIPTION
**Pull request by juliusknorr**
I cannot test this, but according to #10052 and https://www.campaignmonitor.com/css/email-client/ibm-notes-9/ it should work. :see_no_evil: 

@michag86 Can you maybe test this and prove that it works?

### CODE REVIEW

**Comment by michag86** [comment]
Okay, I will test this.

**Comment by MorrisJobke** [comment]
That also means that any possible changed templates out there will not work anymore, because of the changed order of parameters.

**Comment by MorrisJobke** [comment]
@juliushaertl Could you fix the unit tests as you changed the result HTML

**Comment by juliushaertl** [comment]
> That also means that any possible changed templates out there will not work anymore, because of the changed order of parameters.

Ah so the custom templates only change the template strings :/ We should consider moving this to some template engine then, so we can have proper named parameters in the template string.

**Comment by MorrisJobke** [comment]
> Ah so the custom templates only change the template strings :/ We should consider moving this to some template engine then, so we can have proper named parameters in the template string. (maybe something for 15)

Or just use named replacements ;) But yes - something for 15 and we document it just and are done ;)

**Comment by juliushaertl** [comment]
Ok, let me fix the tests...

File: lib/private/Mail/EMailTemplate.php [hunk 1]
```diff
@@ -539,7 +539,7 @@ public function addBodyButtonGroup(string $textLeft,
 		$color = $this->themingDefaults->getColorPrimary();
 		$textColor = $this->themingDefaults->getTextColorPrimary();
 
-		$this->htmlBody .= vsprintf($this->buttonGroup, [$color, $color, $urlLeft, $color, $textColor, $textColor, $textLeft, $urlRight, $textRight]);
+		$this->htmlBody .= vsprintf($this->buttonGroup, [$color, $color, $color, $urlLeft, $color, $textColor, $textColor, $textLeft, $urlRight, $textRight]);
```
Comment (nickvergessen) [inline lib/private/Mail/EMailTemplate.php]: You can simply move the new value to the end and replace all `%s` with their `%1$s` equivalents.
That would keep compatibility

**Comment by juliushaertl** [comment]
Thanks @nickvergessen for the hint :+1: 

Fixed that so we don't break compatibility. The failing test is unrelated.

**Review by nickvergessen — APPROVED** [review]

**Comment by michag86** [comment]
Changes from EMailTemplate.php  are tested and working.

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**